plugin-maven/README.md

```diff
@@ -1021,11 +1021,40 @@ Uses Jackson and YAMLFactory to pretty print objects:
 </jackson>
 ```
 
+## Shell
+
+- `com.diffplug.spotless.maven.FormatterFactory.addStepFactory(FormatterStepFactory)` [code](./src/main/java/com/diffplug/spotless/maven/shell/Shell.java)
+
+```xml
+<configuration>
+  <shell>
+    <includes>     <!-- Not required. Defaults to **/*.sh -->
+      <include>scripts/**/*.sh</include>
+    </includes>
+    
+    <shfmt />    <!-- has its own section below -->
+  </shell>
+</configuration>
+```
+
+### shfmt
+
+[homepage](https://github.com/mvdan/sh). [changelog](https://github.com/mvdan/sh/blob/master/CHANGELOG.md).
+
+When formatting shell scripts via `shfmt`, configure `shfmt` settings via `.editorconfig`.
```

Comment:
If we can note a bit more here, that the `.editorconfig` file only be supported at the same level as shell files, you can't override it's path due to https://github.com/mvdan/sh/issues/1055.

Reply:
That is an incorrect assumption, surprisingly. If I were to have the following project structure, all shell files will respect the `.editorconfig` settings. I can write tests to validate and solidify this claim, but this is the behavior I've seen in a standalone test project. `shfmt` tries to be smart when looking for an `.editorconfig`. I'm not sure I know what the process for finding the `.editorconfig` is, but the developer's https://github.com/mvdan/editorconfig library is used instead of the default Go library and I think that plays a factor in finding the editorconfig.

```
repository-name/
├── client/
│   └── scripts/
│       ├── foobar.sh
│       └── barfoo.sh
├── scripts/
│   ├── foo.sh
│   └── bar.sh
├── .editorconfig
├── init.sh
└── build.gradle.kts
```